pdns/rfc2136handler.cc

```diff
@@ -398,7 +399,17 @@ uint PacketHandler::performUpdate(const string &msgPrefix, const DNSRecord *rr,
           recordsToDelete.push_back(rec);
       }
       if (rr->d_class == QClass::NONE) { // 3.4.2.4
-        if (rrType == rec.qtype && rec.getZoneRepresentation() == rr->d_content->getZoneRepresentation())
+        auto repr = rec.getZoneRepresentation();
+        if (rec.qtype == QType::TXT) {
```

Comment:
Just a performance nit, but would it make sense to skip that dance when `rrType != rec.qtype` (although I'm not sure that actually happens)?

Reply:
can't happen on this path, indeed; however, it's a very good idea for the bigger followup PR I want to land in 4.8 because it'll touch more code paths